numpy/random/mtrand/mtrand.pyx

```diff
@@ -751,15 +777,22 @@ cdef class RandomState:
         """
         cdef ndarray obj "arrayObject_obj"
         cdef int pos
-        algorithm_name = state[0]
-        if algorithm_name != 'MT19937':
-            raise ValueError("algorithm must be 'MT19937'")
-        key, pos = state[1:3]
-        if len(state) == 3:
+        if isinstance(state[0], basestring):
+            version = 0
+            rng, *rest = state
+        else:
+            version, rng, *rest = state
+        if version not in range(_HIGHEST_VERSION + 1):
+            raise ValueError("`version` must be an integer between 0 and {0}".
+                             format(_HIGHEST_VERSION))
```

Comment:
This error should be split in two, to distinguish "this is not a valid version" and "this version is newer than this version of numpy"